src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,38 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
```

Comment:
Consider creating a local variable for the machines list instead of creating new ArrayLists in each return statement. This will make the code cleaner and follow the checklist guideline.